Libraries/Text/TextInput/RCTBaseTextInputView.m

```diff
@@ -619,6 +610,38 @@ - (void)handleInputAccessoryDoneButton
 
 #pragma mark - Helpers
 
+- (void)trimInputTextInRange:(NSRange)range
```

Comment:
Not exactly.
This separate function is still a method (it mutates `self`) which does not solve the complexity problem.

Reply:
@shergin Yeah, actually, Swift already solved this glyphs issue, but OC, we seems need to do these heavy things by ourself. Have no idea how to move on. 😂

Reply:
Yeah, Swift's strings are character-aware but normal Objective-C are not. Yeah... but that's only String's properties, that is not related to UITextField and UITextView implementation, right? So, we already have two subclasses for those classes that enhance the behaviors and fix some issue. Maybe we can encapsulate that character-aware logic there?

Or, what if we move that `maxLength` feature implementation into those classes and try to make it totally transparent for all external logic?